SiftScience/sift-android
# Allow sending mobile sdk events to multiple SIFT instances

### Purpose
Allow sending mobile sdk events to multiple SIFT accounts. Example scenario:

There is a global SIFT account & a regional SIFT account (let's say APAC - Asia Pacific). We want to : 

- send mobile sdk events to global SIFT account for non APAC users 
- send mobile sdk events to both global SIFT & APAC SIFT accounts for APAC users.

At the moment, SIFT sdk only allows sending mobile app events to single SIFT account. This PR will make the SDK flexible to allow sending events to multiple SIFT accounts.

### Technical Detail
The changes are made to support backward compatibility so the existing clients will still be able to use single SIFT account just like before:

```
Sift.open(this, new Sift.Config.Builder()
                .withAccountId("ACCOUNT_ID")
                .withBeaconKey("BEACON_KEY")
                .build());
```

Now we allow initializing the SDK like this as well:

```
List<AccountKey> accountKeys = new ArrayList<>();
accountKeys.add(new AccountKey("ACCOUNT_ID_1", "BEACON_KEY_1"));
accountKeys.add(new AccountKey("ACCOUNT_ID_2", "BEACON_KEY_2"));
Sift.open(this, new Sift.Config.Builder()
                .withAccountKeys(accountKeys)
                .build());
```

### Testing details

- Automated tests
    Added unit tests
- Manual testing
    Ran hello-sift app.
    Check the logs of both SIFT accounts to verify events are being sent to both.
